src/__test__/authorize.test.ts

```diff
@@ -101,3 +101,77 @@ describe('authorize - with secret as callback in options', () => {
     })
   })
 })
+
+describe('authorize - with onAuthentication callback in options', () => {
+  let token: string = ''
+  let tokenWrong: string = ''
+
+  beforeEach(async (done) => {
+    jest.setTimeout(15_000)
+    await fixtureStart(
+      async () => {
+        const response = await axios.post('http://localhost:9000/login')
+        token = response.data.token
+        const responseWrong = await axios.post('http://localhost:9000/login-wrong')
+        tokenWrong = responseWrong.data.token
+        done()
+      },
+      {
+        secret: secretCallback,
+        onAuthentication: decodedToken => {
+          if (decodedToken.checkField === true) {
+            return {
+              email: decodedToken.email
+            }
+          } else {
+            throw new Error('Check Field validation failed')
+          }
+        }
+      }
+    )
+  })
+
+  afterEach((done) => {
+    fixtureStop(done)
+  })
+
+  it('should connect the user', (done) => {
+    const socket = io('http://localhost:9000', {
+      auth: { token: `Bearer ${token}` }
+    })
+    socket.on('connect', () => {
+      socket.close()
+      done()
+    })
+  })
+
+  it('should contain user property', (done) => {
+    const socketServer = getSocket()
+    socketServer?.on('connection', (client: any) => {
+      expect(client.user.email).toEqual('john@doe.com')
+    })
+    const socket = io('http://localhost:9000', {
+      auth: { token: `Bearer ${token}` }
+    })
+    socket.on('connect', () => {
+      socket.close()
+      done()
+    })
+  })
+
+  it('should emit error when user validation fails', (done) => {
+    const socket = io('http://localhost:9000', {
+      auth: { token: `Bearer ${tokenWrong}` }
+    })
+    socket.on('connect_error', (err: any) => {
+      try {
+        expect(err.message).toEqual('Check Field validation failed lol')
```

Comment:
The error message should be  `Check Field validation failed` not  `Check Field validation failed lol` right ?

Reply:
Haha, my bad